src/models/necks/segmentation/unet_neck.py

```diff
@@ -0,0 +1,184 @@
+            in_channels + skip_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+
+        self.attention1 = SCSEModule(channels=in_channels + skip_channels) if use_attention else None
+
+        self.conv2 = ConvBnAct(
+            out_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+
+        self.attention2 = SCSEModule(channels=out_channels) if use_attention else None
+
+    def forward(self, x: Tensor, skip: Optional[Tensor] = None) -> Tensor:
+        """Forward method.
+
+        Args:
+            x: Tensor for upsample.
+            skip: Skip-connection features.
+        """
+        x = F.interpolate(x, scale_factor=2, mode="nearest")
+
+        if skip is not None:
+            if skip.size(2) != x.size(2):
+                skip = F.interpolate(skip, size=(x.size(2), x.size(3)), mode="nearest")
+
+            x = torch.cat([x, skip], dim=1)
+
+            if self.attention1 is not None:
+                x = self.attention1(x)
+
+        x = self.conv1(x)
+        x = self.conv2(x)
+
+        if self.attention2 is not None:
+            x = self.attention2(x)
+
+        return x
+
+
+class CenterBlock(nn.Sequential):
+    """Optional building block for Unet like architecture."""
+    def __init__(self, in_channels: int, out_channels: int, use_batchnorm: int = True):
+        """Init CenterBlock.
+
+        Args:
+            in_channels: Input channels.
+            out_channels: Output channels.
+            use_batchnorm: If True then `ConvBnAct` will use batch normalization.
+        """
+        conv1 = ConvBnAct(
+            in_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+        conv2 = ConvBnAct(
+            out_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm)
+        super().__init__(conv1, conv2)
+
+
+@NECKS.register_class
+class UnetNeck(BaseModel):
+    """Unet is a fully convolution neural network for image semantic segmentation.
+    Paper: https://arxiv.org/pdf/1505.04597.
+    """
+
+    def __init__(self,
+                 encoder_channels: Tuple[int],
+                 decoder_channels: Tuple[int] = (512, 256, 128, 64, 32),
+                 n_blocks: int = 5,
+                 use_batchnorm: bool = True,
+                 use_attention: bool = False,
+                 center: bool = True):
+        """Init Unet
+
+        Args:
+            encoder_channels: Numbers of ``Conv2D`` layer filters.
+            decoder_channels: Numbers of ``Conv2D`` layer filters in decoder blocks.
+            n_blocks: number of stages used in decoder, larger depth - more features are generated.
+                e.g. for depth=3 encoder will generate list of features with following spatial shapes
```

Comment:
So is that means that UnetNeck work onlu with backbones with encoder features  (H/(2^depth), W/(2^depth)] ?

Reply:
No, it's just example. You can use any encoder feature shapes.